pkg/operator/staticresource/controller.go

```diff
@@ -0,0 +1,289 @@
+package staticresource
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cloud-credential-operator/pkg/assets/v410_00_assets"
+	"github.com/openshift/cloud-credential-operator/pkg/operator/platform"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcehelper"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	log "github.com/sirupsen/logrus"
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	admissionregistrationclientv1beta1 "k8s.io/client-go/kubernetes/typed/admissionregistration/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	controllerName    = "staticresource"
+	deploymentName    = "cloud-credential-operator"
+	operatorNamespace = "openshift-cloud-credential-operator"
+)
+
+var (
+	staticFiles = []string{
+		"v4.1.0/aws-pod-identity-webhook/sa.yaml",
```

Comment:
Is this v4.1.0 the version of aws-pod-identity-webhook? Seems like the directory structure should be inverted, aws-pod-identity-webhook/v4.1.0/sa.yaml

Reply:
yes

Reply:
actually no.  that is referring to the OCP version.  the idea is that if some point in the future, you need access to both the old asset and the new in order to do some non-trivial reconciliation, you'd have both versions.  not sure if i buy the organization/usefulness of that though.  it's just how the kube-* operators do it. eg https://github.com/openshift/cluster-kube-apiserver-operator/tree/master/bindata/v4.1.0/kube-apiserver